File: src/main/php/Easy/IsBoomerang.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class IsBoomerang
{
    /**
     * @param  int[][]  $points
     */
    public function transformIsBoomerang(array $points): bool
    {
        $distinct = [];
        foreach ($points as [$x, $y]) {
            $key = "{$x},{$y}";
            if (isset($distinct[$key])) {
                return false;
            }
            $distinct[$key] = true;
        }
        if (count($distinct) !== 3) {
            return false;
        }
        [$x1, $y1] = $points[0];
        [$x2, $y2] = $points[1];
        [$x3, $y3] = $points[2];
        $area = abs($x1 * ($y2 - $y3) + $x2 * ($y3 - $y1) + $x3 * ($y1 - $y2));

        return $area !== 0;
    }

    public function solve(array $points): bool
    {
        return $this->transformIsBoomerang($points);
    }
}

File: lib/easy/is_boomerang.dart

bool isBoomerang(List<List<int>> points) {
  // Check for distinct points
  Set<String> distinct = points.map((point) => point.join(',')).toSet();
  if (distinct.length != 3) return false;

  // Extract coordinates
  int x1 = points[0][0];
  int y1 = points[0][1];
  int x2 = points[1][0];
  int y2 = points[1][1];
  int x3 = points[2][0];
  int y3 = points[2][1];

  // Check if points are non-collinear using the area of triangle formula
  return (x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)).abs() != 0;
}
